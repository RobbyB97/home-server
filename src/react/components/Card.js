import React from 'react'
import {Link} from 'react-router-dom'

const Card = ({image, title, subtext, link}) => (
    <div>
        <div className="card__wrap">
            <a className="card__link" href={link}>
                <div className="card">
                    <img src={image}/>
                    <div className="card__text">
                        <h2>{title}</h2>
                        <p>{subtext}</p>
                    </div>
                </div>
            </a>
        </div>
    </div>
)

export default Card