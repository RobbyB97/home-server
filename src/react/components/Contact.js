import React from 'react';


export class Contact extends React.Component {
    render() {
        return (
            <section id="Contact" className="contact">
                <div className="contact__shape one"></div>
                <div className="contact__shape two"></div>
                Contact component
            </section>
        );
    }
}


export default Contact;