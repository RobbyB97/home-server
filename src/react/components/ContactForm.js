/* Libraries */
import React from 'react'


export class ContactForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            nameValue: ''
        };

        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({
            ...this.state,
            nameValue: event.target.value
        });
    }

    handleSubmit(event) {
        console.log('hi')
    }

    render() {
        return (
            <div className="ContactForm">
                <br /><br /><br /><br /><br />
                <form id="contactForm">
                    <p>
                        <label>Name</label>
                        <input type="text" 
                            value={this.state.nameValue} 
                            onChange={this.handleNameChange} 
                            required 
                        />
                    </p>
                </form>
            </div>
        );
    }
}


export default ContactForm