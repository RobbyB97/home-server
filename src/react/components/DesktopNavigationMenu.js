import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {setTitle} from '../actions/title'


const DesktopNavigationMenu = ({dispatch}) => (
    <div id="desktop-navigation-menu">
        <ul>
            <li>
                <NavLink to="/react/" onClick={() => {
                    dispatch(setTitle(''))
                }}>Home</NavLink>
            </li>
            <li>
                <NavLink to="/react/style_guide" onClick={() => {
                    dispatch(setTitle('Style Guide'))
                }}>Style Guide</NavLink>
            </li>
        </ul>
    </div>
)

export default connect()(DesktopNavigationMenu)