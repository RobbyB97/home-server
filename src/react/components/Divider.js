import React from 'react';


export class Divider extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className={`divider ${this.props.section}`}>
                <div className="divider__shape one"></div>
                <div className="divider__shape two"></div>
            </div>
        );
    }
}


export default Divider;