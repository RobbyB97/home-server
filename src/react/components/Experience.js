import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                <div className="experience__shape one"></div>
                <div className="experience__shape two"></div>
                Experience component
            </section>
        );
    }
}


export default Experience;