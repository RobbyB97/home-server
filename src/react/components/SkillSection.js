import React from 'react';


export class SkillSection extends React.Component {
    render() {
        return (
            <article className="skillSection">
                <h2 className="skillSection__title">
                    {this.props.title}
                </h2>

                <ul className="skillSection__list">
                    {this.props.skills && 
                        this.props.skills.map((skill) => (
                            <li key={skill}>
                                {skill}
                            </li>
                        ))
                    }
                </ul>
            </article>
        );
    }
}


export default SkillSection;