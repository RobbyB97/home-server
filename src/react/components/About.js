import React from 'react';

import Divider from './Divider'


export class About extends React.Component {
    render() {
        return (
            <section id="About" className="about">

                <div className="about__content one">
                    <div className="about__portrait">
                        <img src="/dist/images/aboutImage.png" />
                        <div className="about__text one">
                            <p>
                                <span className="one">Hi, my name's Robby. <span className="two">I'm a front-end developer and digital artist</span>.</span>
                            </p>
                        </div>
                    </div>
                </div>

                <Divider section="about two" />

                <div className="about__content two">
                    <div className="about__text two">
                        <p>
                            I want to help you succeed by building a beautiful, fast and functional website for your business.
                        </p>
                    </div>
                </div>
            </section>
        );
    }
}


export default About;