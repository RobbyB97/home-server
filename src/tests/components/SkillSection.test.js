import React from 'react';
import { shallow } from 'enzyme';

import { SkillSection } from '../../react/components/SkillSection';
import skillSections from '../fixtures/skillSections';


test('Render SkillSection without props', () => {
    const wrapper = shallow(<SkillSection />);
    expect(wrapper).toMatchSnapshot();
});

test('Render SkillSection with title', () => {
    const wrapper = shallow(<SkillSection {...skillSections[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render SkillSection with title and skills', () => {
    const wrapper = shallow(<SkillSection {...skillSections[1]}/>);
    expect(wrapper).toMatchSnapshot();
});