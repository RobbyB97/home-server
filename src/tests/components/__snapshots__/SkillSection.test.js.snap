// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SkillSection with title 1`] = `
<article
  className="skillSection"
>
  <header
    className="skillSection__header"
  >
    <h2
      className="skillSection__title"
    >
      Title!
    </h2>
  </header>
  <section
    className="skillSection__list"
    data-active={true}
  >
    <ul />
  </section>
  <footer
    className="skillSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render SkillSection with title and skills 1`] = `
<article
  className="skillSection"
>
  <header
    className="skillSection__header"
  >
    <h2
      className="skillSection__title"
    >
      Title!
    </h2>
  </header>
  <section
    className="skillSection__list"
    data-active={true}
  >
    <ul>
      <li
        key="Skill1"
      >
        Skill1
      </li>
      <li
        key="Skill2"
      >
        Skill2
      </li>
      <li
        key="Skill3"
      >
        Skill3
      </li>
    </ul>
  </section>
  <footer
    className="skillSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render SkillSection without props 1`] = `
<article
  className="skillSection"
>
  <header
    className="skillSection__header"
  >
    <h2
      className="skillSection__title"
    />
  </header>
  <section
    className="skillSection__list"
    data-active={true}
  >
    <ul />
  </section>
  <footer
    className="skillSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;
